src/dev-app/youtube-player/youtube-player-demo.ts

```diff
@@ -33,8 +35,30 @@ const VIDEOS: Video[] = [
   templateUrl: 'youtube-player-demo.html',
   styleUrls: ['youtube-player-demo.css'],
 })
-export class YouTubePlayerDemo {
-  video: Video | undefined = VIDEOS[0];
+export class YouTubePlayerDemo implements AfterViewInit, OnDestroy {
+  @ViewChild('demoYouTubePlayer') demoYouTubePlayer: ElementRef<HTMLDivElement>;
+  selectedVideo: Video | undefined = VIDEOS[0];
   videos = VIDEOS;
-  apiLoaded = false;
+  videoWidth: number | undefined;
+  videoHeight: number | undefined;
+
+  constructor(private _changeDetectorRef: ChangeDetectorRef) {
+    this.onResize = this.onResize.bind(this);
```

Comment:
Instead of re-assigning a method, you can change it to be an arrow function which lets you avoid the `bind`. E.g. `onResize = () => {...}`.

Reply:
Ah yeah, I had messed with that syntax before but it gave me some issues when removing the event listener. I've updated to use this and solved the event listener removal issue. Thanks.